### DISCUSSION TITLE
Is there a way to toggle sort?

### DISCUSSION THREAD

**Discussion by akulmehta** [spatie/laravel-query-builder#548] [Q&A] [answered]
This may be a very naive question but, I want to have an icon in a table which allows users to click on it and toggle the sort order.

Example: the icon next to the table header "ID" when clicked should toggle the sort for that field.
![image](https://user-images.githubusercontent.com/42812604/96288880-1d979200-0fb2-11eb-854f-77278d1a78b0.png)

So how can I create a toggle function to toggle the sort when a user clicks on that button/icon?

Before using this library I was producing the table using blade as the following to also pass the params to the controller.
```
                <th class="col-1 text-center">
                    ID
                    <a href="{{route('posts.index', 
                        [
                        'togglesort_id' => true, 
                        ] + $params)}}">
                        <i class="fas fa-sort"></i>
                    </a>
                </th>
```
Then in the controller if the togglesort_id param is true I would change the `orderby` for the query.

But there must be a more elegant way of doing this using this library. I am just not sure what. Any help is appreciated.

**Comment by akulmehta** [comment]
I am really stuck on this still. Any help or alternate mechanisms to tackle this is appreciated. e.g. change the blade to contain an if statement to produce the icon in the view dynamically?

**Comment by dominikb** [comment, marked as answer]
For information on how to sort see the docs: [QueryBuilder Sorting](https://spatie.be/docs/laravel-query-builder/v3/features/sorting)

For sort-dependent rendering, you can fall back to using the underlying `QueryBuilderRequest`:

```php
app(QueryBuilderRequest::class)->sorts()->contains('id'); // ascending
app(QueryBuilderRequest::class)->sorts()->contains('-id'); // descending
app(QueryBuilderRequest::class)->sorts()->contains(function($value, $key) {
    return in_array($value, ['-id', 'id']);
}); // no query defined sort for id was applied
```

You can query for the sort order directly in your blade template or save it in the context and pass it to the blade template.
From there you could use an if-else construct to dynamically apply classes or render icons.

**Reply by akulmehta** [reply]
Thanks I got it to work and this helps.
